### PR TITLE
NXP-25507: Add WorkManagerConfiguration to CoreFeature

### PR DESCRIPTION
**Pull request by ron1**

### CODE REVIEW

File: nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/WorkManagerConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        if (WORK_MANAGER_KAFKA_STREAM.equals(workManagerType)) {
+            initKafka();
+        }
+    }
+
+    protected String calcWorkManagerType() {
+        final String value = System.getProperty(WORK_MANAGER_PROPERTY);
+        if (StringUtils.isNotEmpty(value)) {
+            if (Stream.of(WORK_MANAGER_DEFAULT, WORK_MANAGER_CHRONICLE_STREAM, WORK_MANAGER_KAFKA_STREAM).noneMatch(
+                    s -> s.equals(value))) {
```
Comment (efge) [inline nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/WorkManagerConfiguration.java]: No need for streams:

    if (!Arrays.asList(WORK_MANAGER_DEFAULT, WORK_MANAGER_CHRONICLE_STREAM, WORK_MANAGER_KAFKA_STREAM).contains(value))

File: nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/WorkManagerConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            workManagerType = WORK_MANAGER_DEFAULT;
+        }
+        log.debug(String.format(
+                "Calculated workManagerType value to be '%s' since System Property '%s' was not specified",
+                workManagerType, WORK_MANAGER_PROPERTY));
+        return workManagerType;
+    }
+
+    protected boolean isKafkaEnabled() {
+        return "true".equals(System.getProperty("kafka")) && KafkaUtils.kafkaDetected();
```
Comment (efge) [inline nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/WorkManagerConfiguration.java]: Please put `"kafka"` in a constant. And could we use something longer like `nuxeo.test.kafka.enabled`, or is this already something used by other parts of the tests?

**Comment by efge** [comment]
@bdelbosc @kevinleturc please review as well

File: nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/WorkManagerConfiguration.java (hunk 1 above)
Comment (ron1) [reply nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/WorkManagerConfiguration.java]: Agreed

File: nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/WorkManagerConfiguration.java (hunk 2 above)
Comment (efge) [reply nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/WorkManagerConfiguration.java]: Ok thanks so it's worth extracting that in a constant used in both locations for clarity please.

**Comment by efge** [comment]
@ron1 Nuxeo will take this PR and do a few more refactorings before integrating it, no need to update it yourself.

**Comment by kevinleturc** [comment]
Hi @ron1 
Feature was introduced into Nuxeo with #2334, I'm closing this PR.

**Pull request closed without merging**